bug: --border-radius custom css style is not applied to left hand corners of controls

### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

Setting `--border-radius:30px;` in vue.js scoped css  does not apply the full radius to the left hand side of ion-input, ion-select, and ion-textarea. Resulting in the following shapes (respectively):
![image](https://user-images.githubusercontent.com/9688384/229952214-6e106879-81bf-4b88-90f4-fbaaea93d293.png)


### Expected Behavior

Setting `--border-radius:30px;` in the css should render an input similar to this image:
![image](https://user-images.githubusercontent.com/9688384/229949697-aa157f36-d911-4867-9822-67c9114c2e5e.png)


### Steps to Reproduce

Start a new blank ionic-vue project, and replace the content of `src/views/HomePage.vue` with the following content:
```
<template>
  <ion-page>
    <ion-content :fullscreen="true">
      <ion-grid style="max-width: 500px;">
        <ion-row>
          <ion-col style="padding: 20px;">
            <ion-input aria-label="" class="custom-input" fill="outline"></ion-input> 
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col style="padding: 20px;">
            <ion-select class="custom-select" aria-label="" fill="outline">
              <ion-select-option v-for="[value, text] in choices" :value="value" :key="value">{{ text }}</ion-select-option>
            </ion-select> 
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col style="padding: 20px;">
            <ion-textarea class="custom-textarea" aria-label="" fill="outline"></ion-textarea> 
          </ion-col>
        </ion-row>
      </ion-grid>
    </ion-content>
  </ion-page>
</template>

<script setup lang="ts">
import { IonPage, IonContent, IonInput, IonGrid, IonRow, IonCol, IonSelect, IonSelectOption, IonTextarea } from '@ionic/vue';

const choices = new Map([
  ['1', 'Choice 1'],
  ['2', 'Choice 2'],
  ['3', 'Choice 3'],
  ['4', 'Choice 4'],
]);

</script>

<style scoped>
  ion-input.custom-input{
    --border-radius: 30px; /* does not apply to top left and bottom left corners */
    --border-width: 2px;
  }

  ion-select.custom-select{
    --border-radius: 30px; /* does not apply to top left and bottom left corners */
    --border-width: 2px;
  }

  ion-textarea.custom-textarea{
    --border-radius: 30px; /* does not apply to top left and bottom left corners */
    --border-width: 2px;
  }

</style>
```
Run the Created project.

### Code Reproduction URL

https://github.com/mp035/ionic_2023_04_05

### Ionic Info

[```
WARN] Error loading @capacitor/ios package.json: Error: Cannot find module '@capacitor/ios/package'
       
       Require stack:
       - /home/mark/.nvm/versions/node/v16.19.0/lib/node_modules/@ionic/cli/lib/project/index.js
       - /home/mark/.nvm/versions/node/v16.19.0/lib/node_modules/@ionic/cli/lib/index.js
       - /home/mark/.nvm/versions/node/v16.19.0/lib/node_modules/@ionic/cli/index.js
       - /home/mark/.nvm/versions/node/v16.19.0/lib/node_modules/@ionic/cli/bin/ionic
[WARN] Error loading @capacitor/android package.json: Error: Cannot find module '@capacitor/android/package'
       
       Require stack:
       - /home/mark/.nvm/versions/node/v16.19.0/lib/node_modules/@ionic/cli/lib/project/index.js
       - /home/mark/.nvm/versions/node/v16.19.0/lib/node_modules/@ionic/cli/lib/index.js
       - /home/mark/.nvm/versions/node/v16.19.0/lib/node_modules/@ionic/cli/index.js
       - /home/mark/.nvm/versions/node/v16.19.0/lib/node_modules/@ionic/cli/bin/ionic

Ionic:

   Ionic CLI       : 7.0.0 (/home/mark/.nvm/versions/node/v16.19.0/lib/node_modules/@ionic/cli)
   Ionic Framework : @ionic/vue 7.0.0

Capacitor:

   Capacitor CLI      : 4.7.3
   @capacitor/android : not installed
   @capacitor/core    : 4.7.3
   @capacitor/ios     : not installed

Utility:

   cordova-res : not installed globally
   native-run  : 1.7.2

System:

   NodeJS : v16.19.0 (/home/mark/.nvm/versions/node/v16.19.0/bin/node)
   npm    : 8.19.3
   OS     : Linux 5.10


```

### Additional Information

_No response_

--- a/core/src/components/input/input.scss
+++ b/core/src/components/input/input.scss
@@ -24,7 +24,7 @@
    * @prop --highlight-color-invalid: The color of the highlight on the input when invalid
    *
    * @prop --border-color: Color of the border below the input when using helper text, error text, or counter
-   * @prop --border-radius: Radius of the input
+   * @prop --border-radius: Radius of the input. A large radius may display unevenly when using fill="outline"; if needed, use shape="round" instead or increase --padding-start.
    * @prop --border-style: Style of the border below the input when using helper text, error text, or counter
    * @prop --border-width: Width of the border below the input when using helper text, error text, or counter
    */

--- a/core/src/components/select/select.scss
+++ b/core/src/components/select/select.scss
@@ -19,7 +19,7 @@
    * @prop --highlight-color-valid: The color of the highlight on the select when valid
    *
    * @prop --border-color: Color of the select border
-   * @prop --border-radius: Radius of the select border
+   * @prop --border-radius: Radius of the select border. A large radius may display unevenly when using fill="outline"; if needed, use shape="round" instead or increase --padding-start.
    * @prop --border-style: Style of the select border
    * @prop --border-width: Width of the select border
    *

--- a/core/src/components/textarea/textarea.scss
+++ b/core/src/components/textarea/textarea.scss
@@ -9,7 +9,7 @@
    *
    * @prop --border-radius: Border radius of the textarea
    * @prop --border-color: Color of the border below the textarea when using helper text, error text, or counter
-   * @prop --border-radius: Radius of the textarea border
+   * @prop --border-radius: Radius of the textarea border. A large radius may display unevenly when using fill="outline"; if needed, use shape="round" instead or increase --padding-start.
    * @prop --border-style: Style of the border below the textarea when using helper text, error text, or counter
    * @prop --border-width: Width of the border below the textarea when using helper text, error text, or counter
    *

